init_venv.sh

```diff
@@ -69,45 +83,53 @@ else
   fi
 fi
 
-pip install wheel
-
 CONSTRAINTS_FILE=$(tempfile)
-echo numpy==${NUMPY_VERSION} >> ${CONSTRAINTS_FILE}
+cat constraints.txt >> ${CONSTRAINTS_FILE}
```

Comment:
Since precise URI  to install `torch` and `torchvision` depends on the `$CUDA_VERSION_CODE`, I think that temporary file is the best decision.
And IMHO it is really convenient in the situation, when we could point version of some packages in the script.

Reply:
Does that mean, that every call to `pip install <some-package> -c constraints.txt` changes the content of `constraints.txt` file inplace? If no, I see no places where you put something in the constraints file.

Reply:
No, `pip install <some-package> -c constraints.txt` does not change the content of constraints.txt file inplace.
But the script `init_venv.sh` uses a temporary file `${CONSTRAINTS_FILE}`,  then the scripts makes 
```
cat constraints.txt >> ${CONSTRAINTS_FILE}
```
and then, below, there is the following code
```
  echo torch==${TORCH_VERSION}+cpu >> ${CONSTRAINTS_FILE}
  echo torchvision==${TORCHVISION_VERSION}+cpu >> ${CONSTRAINTS_FILE}
  pip install torch==${TORCH_VERSION}+cpu torchvision==${TORCHVISION_VERSION}+cpu -f https://download.pytorch.org/whl/torch_stable.html \
          -c ${CONSTRAINTS_FILE} || exit 1
```
This block of code is repeated with some changes three times inside if-elif-else clause (if there is no CUDA, if `$CUDA_VERSION_CODE == "102" `, and `else` branch for other cases). As the result, the constraints file that is used contains both our constraints from `constraints.txt` and torch/torchvision constraints.
It is very useful to check if our constraints contains conflicts (e.g. pytorch vs numpy, etc) -- in this case any pip operation will cause exit.

Reply:
Ok, thanks. Haven't noticed those lines.